keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components/README.md

```diff
@@ -0,0 +1,239 @@
+
+Large number of features added to klog has lead to large drop in quality. 
+[#90804](https://github.com/kubernetes/kubernetes/pull/90804) shows example 
+where kubeup (canonical way to deploy kubernetes for testing) could not use 
+klog feature to write log files due to scalability issues. The maintainers of 
+klog decided it's easier to re-implementing a canonical klog feature in external
+project than debugging the underlying problem.
+
+As for what configuration options should be standardized for all logging formats
+I would look into 12 factor app standard (https://12factor.net/). It defines
+logs as steams of events and discourages applications from taking on
+responsibility for log file management, log rotation and any other processing
+that can be done externally. This is something that Kubernetes already
+encourages by collecting stdout and stderr logs and making them available via
+kubectl logs. It's somewhat confusing that K8s components don't comply to K8s
+best practices.
+
+### Goals
+
+* Unblock development of alternative logging formats
+* Improve quality of logging in k8s core components
+* Logging should use configuration mechanism developed by WG component-standard
+
+### Non-Goals
+
+* Change klog output format
+
+## Proposal
+
+I propose that Kubernetes core components (kube-apiserver, kube-scheduler,
+kube-controller-manager, kubelet) should drop flags that extend
+logging over events streams or klog specific features. This change should be 
+scoped to only those components and not affect broader klog community.
+
+Flags that should be deprecated:
+
+* --log-dir, --log-file, --log-flush-frequency - responsible for writing to 
+  files and syncs to disk. 
+  Motivation: Remove complexity to make alternative loggers easier to implement
+  and reducing feature surface to improve quality.
+* --logtostderr, --alsologtostderr, --one-output, --stderrthreshold - 
+  responsible enabling/disabling writing to stderr (vs file). 
+  Motivation: Not needed if writing to files is removed.
+* --log-file-max-size, --skip-log-headers - responsible configuration of file 
+  rotation. 
+  Motivation: Not needed if writing to files is removed.
+* --add-dir-header, --skip-headers - klog format specific flags . 
+  Motivation: don't apply to other log formats
+* --log-backtrace-at - an legacy glog feature. 
+  Motivation: No trace of anyone using this feature.
+
+Flag deprecation should comply with standard k8s policy and require 3 releases before removal.
+
+This leaves that two flags that should be implemented by all log formats
+
+* -v - control global log verbosity of Info logs
+* --vmodule - control log verbosity of Info logs on per file level
+
+Those flags were chosen as they have direct effect of which logs are written, 
+directly impacting log volume and component performance.
+
+### User Stories
+
+#### Writing logs to files
+
+We should use go-runner as a official fallback for users that want to retain 
```

Comment:
I've never heard of go-runner, is it used by any Kubernetes components/CI currently? CI uses systemd-run, at least in the node e2es.

Edit: aha, it's used in the scalability tests.

Reply:
ack